components/eamxx/src/control/intensive_observation_period.cpp

```diff
@@ -0,0 +1,329 @@
+  EKAT_REQUIRE_MSG(m_lat_lon_info.count(grid_name) > 0,
+                   "Error! Attempting to read IOP initial conditions on "
+                   +grid_name+" grid, but m_lat_lon_info entry has not been created.\n");
+
+  auto io_grid = m_io_grids[grid_name];
+  if (grid_name=="Physics GLL" && file_has_dim(file_name,"ncol_d")) {
+    // If we are on GLL grid, and nc file contains "ncol_d" dimension,
+    // we need to reset COL dim tag
+    using namespace ShortFieldTagsNames;
+    auto grid = io_grid->clone(io_grid->name(),true);
+    grid->reset_field_tag_name(COL,"ncol_d");
+    io_grid = grid;
+  }
+
+  // Create vector of fields with correct dimensions to read from file
+  std::vector<Field> fields;
+  for (size_t i=0; i<field_names_nc.size(); ++i) {
+    const auto& nc_name    = field_names_nc[i];
+    const auto& eamxx_name = field_names_eamxx[i];
+
+    auto fm_field = eamxx_name!=nc_name
+                    ?
+                    field_mgr->get_field(eamxx_name).alias(nc_name)
+                    :
+                    field_mgr->get_field(eamxx_name);
+    auto fm_fid = fm_field.get_header().get_identifier();
+    auto dims = fm_fid.get_layout().dims();
+    EKAT_REQUIRE_MSG(fm_fid.get_layout().tag(0)==FieldTag::Column,
+                     "Error! IOP inputs read from IC/topo file must have Column "
+                     "as first dim tag.\n");
+
+    // Set first dimension to match input file
+    dims[0] = io_grid->get_num_local_dofs();
+    FieldLayout dp_fl(fm_fid.get_layout().tags(), dims);
+    FieldIdentifier dp_fid(fm_fid.name(), dp_fl, fm_fid.get_units(), io_grid->name());
+    Field dp_field(dp_fid);
+    dp_field.allocate_view();
+    fields.push_back(dp_field);
+  }
+
+  // Read data from file
+  AtmosphereInput dp_ic_reader(file_name,io_grid,fields);
+  dp_ic_reader.read_variables();
+  dp_ic_reader.finalize();
+
+  // For each field, broadcast data from closest lat/lon column to all processors
+  // and copy data into each field's column.
+
+  // Find the maximum size of transfer data over fields.
+  int max_data_size = 0;
+  for (auto& f : fields) {
+    const auto fl = f.get_header().get_identifier().get_layout();
+    if (fl.size()/fl.dim(0) > max_data_size) max_data_size = fl.size()/fl.dim(0);
+  }
+
+  Real transfer_data[max_data_size];
+  for (size_t i=0; i<fields.size(); ++i) {
+    const auto f = fields[i];
+    const auto fl = f.get_header().get_identifier().get_layout();
+    const auto field_rank = fl.rank();
+    const auto fname = field_names_eamxx[i];
+
+    // If I am the rank with the minimum lat/lon distance,
+    // store data of values to broadcast.
+    const auto my_rank = m_comm.rank();
+    const auto col_mpi_rank = m_lat_lon_info[grid_name].mpi_rank_of_closest_column;
+    if (my_rank == col_mpi_rank) {
+      const auto col_idx = m_lat_lon_info[grid_name].local_column_index_of_closest_column;
+      switch (field_rank) {
```

Comment:
These fields were just created above. Since none of these is a subfield of another field, you actually _know_ that they are stored contiguously in memory. Therefore, you can simply use the field's internal pointers for MPI. Something like
```
for (const auto& f : fields) {
  const auto& fl = f.get_header().get_identifier().get_layout();
  const auto col_mpi_rank = m_lat_lon_info[grid_name].mpi_rank_of_closest_column;
  m_comm.broadcast(f.get_internal_view_data<Real,Host>(), fl.size(), col_mpi_rank);
}
```

Reply:
Let me know what you think.

We can't be guaranteed that all ranks own the same amount of columns, so for the mpi rank that owns the column data, I transferred that data to the first column, then broadcasted the first column of data to all the other rank's first column, then copied to field manager data.

Reply:
Ah, yeah. You can't even assume that the other ranks own any column at all... You could create a one-col field, and bcast that, then loop over ncols and copy. Like this:
```
auto col_layout = fl.strip_dim(COL);
FieldIdentifier fid_tmp("temp",col_layout,dummy_units,"");
Field single_col (fid_tmp);
single_col.allocate_view();
if (comm.rank()==mpi_rank_of_closest_col) {
  single_col.deep_copy(f.subfield(0,index_of_closest_col),Host);
}
comm.broadcast(single_col.get_internal_view_data<Real,Host>(),col_layout.size(),mpi_rank_of_closest_col);
for (int icol=0; icol<fl.dim(COL); ++icol) {
   f.subfield(0,icol).deep_copy(single_col,Host);
}
f.sync_to_dev();
```

Reply:
I didn't think about subfielding and using deep_copy. I'll update, this is much nicer than the case approach I'm using.

Reply:
Update it. The only hiccup was using Field::deep_copy with subfield of Column-only fields, but I just added a `if (rank==1)` case when copying to and from `col_data` field.